runtime/compilers/rillv1/parse_api.go

```diff
@@ -77,6 +77,16 @@ func (p *Parser) parseAPI(node *Node) error {
 	}
 	node.Refs = append(node.Refs, resolverRefs...)
 
+	securityRules, err := tmp.Security.Proto()
+	if err != nil {
+		return fmt.Errorf("failed to parse security rules: %w", err)
+	}
+	for _, rule := range securityRules {
+		if rule.GetAccess() == nil {
+			return fmt.Errorf("the 'custom api' resource type only supports 'access' security rules")
```

Comment:
nit: for normal users they just call them `api`, e.g. `type: api` (it's only internally we words like "custom" or "user-defined" to disambiguate from internal/built-in APIs)

```suggestion
			return fmt.Errorf("the 'api' resource type only supports 'access' security rules")
```